Bottom: 363b4fc01b9f8b56bc5e05642ba7f8f62e8a4298
Top:    42a2491d83ba041a57114b79228b5b05ff5e7581
Author: Ian Kent <raven@themaw.net>
Date:   2013-08-02 16:49:59 +0800

Refresh of autofs.debug

---

diff --git a/modules/lookup_nisplus.c b/modules/lookup_nisplus.c
index d783396..b3b44c1 100644
--- a/modules/lookup_nisplus.c
+++ b/modules/lookup_nisplus.c
@@ -614,15 +614,17 @@ int lookup_mount(struct autofs_point *ap, const char *name, int name_len, void *
 		time_t now = time(NULL);
 		int rv = CHE_OK;
 
-		cache_writelock(mc);
-		me = cache_lookup_distinct(mc, key);
-		if (!me)
-			rv = cache_update(mc, source, key, NULL, now);
-		if (rv != CHE_FAIL) {
+		/* Don't update negative cache when re-connecting */
+		if (!(ap->flags & MOUNT_FLAG_REMOUNT)) {
+			cache_writelock(mc);
 			me = cache_lookup_distinct(mc, key);
-			me->status = time(NULL) + ap->negative_timeout;
+			if (!me)
+				rv = cache_update(mc, source, key, NULL, now);
+			if (rv != CHE_FAIL) {
+				me = cache_lookup_distinct(mc, key);
+				me->status = time(NULL) + ap->negative_timeout;
+			cache_unlock(mc);
 		}
-		cache_unlock(mc);
 		free(mapent);
 		return NSS_STATUS_TRYAGAIN;
 	}
